std/utf.d

```diff
@@ -4386,3 +4387,12 @@ if (isSomeChar!C)
     "𐐷".byUTF!wchar().equal([0xD801, 0xDC37]);
     "𐐷".byUTF!dchar().equal([0x00010437]);
 }
+
+///
+unittest
+{
+    import std.algorithm.comparison : equal;
+    import std.exception : assertThrown;
+
+    assertThrown!UTFException("hello0xF0betty".byUTF!(dchar, UseReplacementDchar.no).equal("hellobetty"));
```

Comment:
don't you need to do `"hello\xF0betty"`?

Reply:
No, because it won't complete.

What I'd like to know is why the exception isn't being thrown.

Reply:
... because `0`, `x` and `f` are all valid unicode characters and not e.g. random garbage that can't be converted from utf8 to 32?

Reply:
Dang it, you're right again. I should just give up and go to bed.